docs/v0.6/en/user-guide/client-libraries/go.md

```diff
@@ -36,153 +37,256 @@ import (
 cfg := greptime.NewCfg("127.0.0.1").
-    WithDatabase("public").      // change to your real database
-    WithPort(4001).              // default port
-    WithAuth("username", "password").            // `Username` and `Password`
-    WithDialOptions(options...). // specify your gRPC dail options
-    WithCallOptions()            // specify your gRPC call options
+    // change the database name to your database name
+    WithDatabase("public").
+    // default port
+    WithPort(4001).
+    // set authentication information
+    WithAuth("username", "password")
+
+client, _ := client.New(cfg)
+```
+%}
+
+{template low-level-object%
+
+```go
+// Construct the table schema for CPU metrics
+cpuMetric, err := table.New("cpu_metric")
+if err != nil {
+    // Handle error appropriately
+}
+
+// Add a 'Tag' column for host identifiers
+cpuMetric.AddTagColumn("host", types.STRING)
+// Add a 'Timestamp' column for recording the time of data collection
+cpuMetric.AddTimestampColumn("ts", types.TIMESTAMP_MILLISECOND)
+// Add 'Field' columns for user and system CPU usage measurements
+cpuMetric.AddFieldColumn("cpu_user", types.FLOAT)
+cpuMetric.AddFieldColumn("cpu_sys", types.FLOAT)
+
+// Insert example data
+// NOTE: The arguments must be in the same order as the columns in the defined schema: host, ts, cpu_user, cpu_sys
+err = cpuMetric.AddRow("127.0.0.1", time.Now(), 0.1, 0.12)
+if err != nil {
+    // Handle error appropriately
+}
 
-client, _ := greptime.NewClient(cfg)
 ```
 
 %}
 
+{template create-rows%
 
+```go
+cpuMetric, err := table.New("cpu_metric")
+if err != nil {
+    // Handle error appropriately
+}
+cpuMetric.AddTagColumn("host", types.STRING)
+cpuMetric.AddTimestampColumn("ts", types.TIMESTAMP_MILLISECOND)
+cpuMetric.AddFieldColumn("cpu_user", types.FLOAT)
+cpuMetric.AddFieldColumn("cpu_sys", types.FLOAT)
+err = cpuMetric.AddRow("127.0.0.1", time.Now(), 0.1, 0.12)
+if err != nil {
+    // Handle error appropriately
+}
 
-{template row-object%
+memMetric, err := table.New("mem_metric")
+if err != nil {
+    // Handle error appropriately
+}
+memMetric.AddTagColumn("host", types.STRING)
+memMetric.AddTimestampColumn("ts", types.TIMESTAMP_MILLISECOND)
+memMetric.AddFieldColumn("mem_usage", types.FLOAT)
+err = memMetric.AddRow("127.0.0.1", time.Now(), 112)
+if err != nil {
+    // Handle error appropriately
+}
+```
+
+%}
 
-The Go ingester SDK uses `Series` to represent a row data item. Multiple `Series` can be added to a `Metric` object and then written to GreptimeDB.
+{template insert-rows%
+
+```go
+resp, err := cli.Write(context.Background(), cpuMetric, memMetric)
+if err != nil {
+    // Handle error appropriately
+}
+log.Printf("affected rows: %d\n", resp.GetAffectedRows().GetValue())
+```
 
 %}
 
+{template streaming-insert%
+
+To insert data using the streaming API, you must first create a stream client.
+
+```go
+cfg := config.New("<host>").
+    WithAuth("<username>", "<password>").
+    WithDatabase(database)
```

Comment:
the code block above is different from unary insert. Is it needed to unify the code sample to:

```go
cfg := config.New("<host>").
    WithAuth("<username>", "<password>").
    WithDatabase("<database>")
```